contracts/core/EntryPoint.sol

```diff
@@ -378,8 +379,9 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
     function getUserOpHash(
         PackedUserOperation calldata userOp
     ) public view returns (bytes32) {
+        bytes32 overrideInitCodeHash = _getEip7702InitCodeHashOverride(userOp);
         return
-            MessageHashUtils.toTypedDataHash(getDomainSeparatorV4(), userOp.hash());
+            MessageHashUtils.toTypedDataHash(getDomainSeparatorV4(), userOp.hash(overrideInitCodeHash));
```

Comment:
`userOp.hash()` may only ever be called with the result of `_getEip7702InitCodeHashOverride()`, which can only be called with the `userOp` as a parameter. The value of `overrideInitCodeHash` is only used once.

Do we need this value to bubble up into the EntryPoint itself? Why not call the `_getEip7702InitCodeHashOverride` directly inside the `userOp.hash`?

Reply:
Separation of concerns: 
UserOperationLib is a library, and its methods (including `hash()`) are pure.
`_getEip7702InitCodeHashOverride()` is a view function, reading the current delegate.